File: app/lib/models/evidence.dart
/// Model representing a piece of evidence submitted during a debate.
class Evidence {
  final String evidenceId;
  final String content;
  final String sourceType;
  final String sourceDetail;
  final String submittedBy;
  final int round;
  final String speaker;

  Evidence({
    required this.evidenceId,
    required this.content,
    required this.sourceType,
    required this.sourceDetail,
    required this.submittedBy,
    required this.round,
    required this.speaker,
  });

  /// Create an Evidence from a JSON map.
  factory Evidence.fromJson(Map<String, dynamic> json) {
    return Evidence(
      evidenceId: json['evidence_id'] as String,
      content: json['content'] as String,
      sourceType: json['source_type'] as String,
      sourceDetail: json['source_detail'] as String,
      submittedBy: json['submitted_by'] as String,
      round: json['round'] as int,
      speaker: json['speaker'] as String,
    );
  }

  /// Convert this Evidence to a JSON map.
  Map<String, dynamic> toJson() {
    return {
      'evidence_id': evidenceId,
      'content': content,
      'source_type': sourceType,
      'source_detail': sourceDetail,
      'submitted_by': submittedBy,
      'round': round,
      'speaker': speaker,
    };
  }
}

File: backend/app/models/evidence.py
"""
Evidence model for LawNuri.

Represents pieces of evidence submitted during a debate,
including uploaded documents, legal statutes, court precedents,
and knowledge graph relations.
"""

from typing import Literal
from uuid import uuid4

from pydantic import BaseModel, Field


class Evidence(BaseModel):
    """A single piece of evidence submitted during a debate round."""

    evidence_id: str = Field(default_factory=lambda: str(uuid4()))
    content: str
    source_type: Literal[
        "uploaded_document",
        "legal_statute",
        "court_precedent",
        "graph_relation",
    ]
    source_detail: str  # e.g., "Patent Act Article 126", "document.pdf p.15"
    submitted_by: str  # team_a or team_b
    round: int
    speaker: str  # agent_id
    relevance: str = ""
